grpc/rust/lib.rs

```diff
@@ -0,0 +1,18 @@
+//
+// Copyright (C) 2022 Vaticle
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+//
+
+tonic::include_proto!("typedb.protocol");
```

Comment:
Why do we use a one-line file here? Is there some way we can merge `lib.rs` and `build.rs` - or is there a specific reason why having both is useful?

Reply:
`rust_library` requires at least a single source file that would contain the library's code. The one-line macro simply includes all the code generated through `protoc`. This way, you can just import the `rust_library` from your code directly.

`build.rs` is the build file that runs before compiling the `rust_library`.

You can't join the files, as the build script runs before compiling the library, and the library relies on the output of the build script.